imap_processing/spice/time.py

```diff
@@ -90,3 +90,75 @@ def _sct2e_wrapper(
         return np.array([spice.sct2e(IMAP_SC_ID, s) for s in sclk_ticks])
     else:
         return spice.sct2e(IMAP_SC_ID, sclk_ticks)
+
+
+@typing.no_type_check
+@ensure_spice
+def str_to_et(
+    time_str: Union[str, list[str]],
```

Comment:
Changing from `list` to ~~`Collection`~~ `Iterable` makes this more generic so that it can accept things like a `ndarray` of strings.
```suggestion
    time_str: Union[str, Iterable[str]],
```
This change will need to be coupled with a flip of the `if` statement to check for a `str` instance instead of `list`.